version: '3.3'

services:
  label_studio:
    restart: on-failure
    image: heartexlabs/label-studio:latest
    container_name: label-studio
    ports:
    - "8080:8080"
    volumes:
    - ${DP_DATA_PATH}/label_studio/pv_project:/label-studio/pv_project
    - ${DP_DATA_PATH}/label_studio/tasks:/label-studio/tasks
    - ./config.xml:/label-studio/config.xml
    stdin_open: true
    tty: true
    command: label-studio start pv_project --init --force --label-config /label_studio/config.xml
  dp_db:
    restart: on-failure
    image: mysql/mysql-server:8.0
    container_name: dp-mlflow-db
    expose:
      - "3306"
    networks:
      - dp_network
    environment:
      - MYSQL_DATABASE=${DP_MYSQL_DATABASE}
      - MYSQL_USER=${DP_MYSQL_USER}
      - MYSQL_PASSWORD=${DP_MYSQL_PASSWORD}
      - MYSQL_ROOT_PASSWORD=${DP_MYSQL_ROOT_PASSWORD}
    volumes:
      - ${DP_DATA_PATH}/dbdata:/var/lib/mysql
  dp_web:
    restart: on-failure
    build:
      context: .
    image: data-programming-image
    container_name: dp-mlflow-server
    ports:
      - "5000:5000"
    networks:
      - dp_network
    environment:
      - MLFLOW_TRACKING_URI=${DP_MLFLOW_TRACKING_URI}
    command: wait-for-it dp-mlflow-db:3306 -s -- mlflow server --backend-store-uri ${DP_MLFLOW_TRACKING_URI} --default-artifact-root ./mlruns --host 0.0.0.0
    stdin_open: true
    tty: true
    volumes:
      - ${DP_DATA_PATH}/mlruns:/code/mlruns
    depends_on:
      - dp_db
  dp:
    build:
      context: .
    image: data-programming-image
    container_name: dp-mlflow
    volumes:
      - ${DP_DATA_PATH}/mlruns:/code/mlruns
      - ${DP_DATA_PATH}/unlabeled_data:/unlabeled_data
      - ./label:/code/label
      - ./MLproject:/code/MLproject
      - ./conda.yaml:/code/conda.yaml
    networks:
      - dp_network
    environment:
      - MLFLOW_TRACKING_URI=${DP_MLFLOW_TRACKING_URI}
      - RESOURCES_PATH=${RESOURCES_PATH}
    stdin_open: true
    tty: true
    depends_on:
      - dp_db

networks:
  dp_network:
    driver: bridge
